### DISCUSSION TITLE
Definition of CG and AERORP using wind tunnel data

### DISCUSSION THREAD

**Discussion by jean33-star** [JSBSim-Team/jsbsim#1124] [Q&A]
Hi, 

I have access to complete wind tunnel datas. They provide a complete characterization of the aerodynamic forces/moments experienced by the aircraft. The data are in the BODY frame,  and  moments are expressed at the balance resolution center. (I know the position of every important points of the aircraft). 

I am wondering how to integrate these data into a JSBSIM model.

As I understand it, I have to:

- Define the CG at the known location for the aircraft (in the STRUCTURAL frame)
- Define the AERORP at the balance resolution center (in the STRUCTURAL frame)
- In the Aerodynamics section, define the axis of expression to BODY 

But, should I also use something like this (seen in another discussion) ? : 
<function name="aero/force/Lift_wbh" **apply_at_cg="true"**>
or it is non necessary ? 

I am very concerned about how the position of the CG affects the moments applied to the aircraft... 

I look forward to your advice, I can provide you more details if necessary

JS

**Comment by seanmcleod** [comment]
I don't see why you would need to use `apply_at_cg`. 

> I am very concerned about how the position of the CG affects the moments applied to the aircraft...

What are your concerns?

This is the code that handles the position of the cg, which also moves as fuel is burnt off etc.

https://github.com/JSBSim-Team/jsbsim/blob/651ad3c78f68d9c0431dd5f984ff9bb8f4d24b8a/src/models/FGAerodynamics.cpp#L288

**Comment by alanjt** [comment]
My understanding is that AERORP It is the reference point about which pitch, roll and yaw moments are calculated,  If the centre of lift is aft of the AERORP, then a nose down pitch moment must be defined.,

I have only recently come across this thread, but also have concerns about the JSBSim documentation.

CG is the actual aircraft centre of gravity.  Unfortunately "CG" is also frequently used to define the origin of the axis system. This causes a lot of confusion.

I also have a full set of aero data from the manufacturer. The manufacturer used several different reference "CG" points in their reports . (note the loose use of CG). This is likely to be due to a variety of reasons, including configuration of the force measuring systems in the various wind tunnels, as well as specific end user data requirements .  So I have transformed all of their derivative and aero coefficient data to be about a single reference point and use that point as my AERORP.

Both AERORP and empty CG are defined relative to the manufacturer´s structural datum. (in my  case this is near the aircraft nose.) Additional weights are added using JSBSim PointMass and fuel contents. Using these JSBSim calculates the actual CG position during flight.  JSBSim copes with all of this and the trim (elevator angle etc) results are very close to the manufacturers data. It also handles the changes in moments of inertia with payload and fuel. 

I find the pictures in the official JSBSim documentation to be confusing. https://jsbsim-team.github.io/jsbsim-reference-manual/user/concepts/frames-of-reference/.   I believe that point G is the AERORP, not the CG. The aircraft weight will act downwards through the actual CG which varies with loading .

Sean - what do you think?

Regards 

Alan

**Comment by agodemar** [comment]
@alanjt CG is the CG. And by definition is also the origin of the standard body-fixed reference frame. 
Ideally, if the aircraft is rigid, if its mass is constant, if its inertias are constant, the 'body frame' motion in space perfectly represents the aircraft dynamics, according to Newton's Laws. In introductory textbooks on Flight Mechanics, those laws are written down first for a constant-mass, rigid-body aircraft. 
But the real picture gets a little bit more complicated when one admits a variable mass. Then one can admit a variable inertia tensor as well. Finally, the rigid-body hypothesis could also fall. Then the CG moves with respect to the airframe, which in turn moves in space.
It's something that should be treated with more advanced mathematical tools and requires a correct integration of what one knows about aeroelasticity, unsteady aerodynamics, and advanced mechanics in general.

Coming to the AERORP: You should think at it as the reference pole used by aerodynamicists to define the aerodynamic moments (or moment coefficients). You cannot provide rolling-, pitching-, and yawing-moment data without specifying the pole. Usually, in wind tunnel testing the pole (or 'reduction point') is the balance centre - the instrument they use to acquire forces and moments. In wind tunnel tests, lab guys do not care about CG, simply because the CG position is not known to them and is determined by the designer in his analyses, in well defined conditions. Lab guys can only assume a hypothetical CG position, for instance, at the balance centre; but this is not necessary. They just provide aerodynamic data by saying: These are the force and moment coefficients "reduced to this pole (x,y,z)" in a given body-fixed frame (their standard definition of body frame is different sometimes, but this is another story).
Finally, we have reality: CG position is only known, really, right before take-off and the pilot can tell where it is within the allowed CG excursion range...

In JSBSim, the moments "about the AERORP" (read "about the given pole") and all the aerodynamic force components are 'transported' or 'reduced' (these terms are taken from classical mechanics) - at each time step - to the CG. This transport is necessary because the flight dynamics equations are implemented assuming that the external action (gravity + aerodynamics + propulsion + particular applied forces) is reduced to the CG.

Hope this helps,
Agostino

**Comment by seanmcleod70** [comment]
Hi Alan, why do you believe that point G in diagrams represents AERORP? By the way, thanks to @agodemar for all pictures/diagrams he contributed.

Mainly a repetition of what you and @agodemar stated, but with an example diagram with some example figures, in case it may help clarify things for you or others reading this discussion at some point.

So let's visualize an aircraft model mounted in a wind tunnel. 

<img width="986" height="306" alt="image" src="https://github.com/user-attachments/assets/3ec923e9-6666-4c99-81b5-3632af0460ac" />

The balance system has been mounted 4m aft of the nose. As @agodemar mentioned, the people doing the wind tunnel testing don't really need to know where the exact CG point is for the model, but let's assume it's where I've indicated, i.e. 7m aft of the nose.

Then for some airspeed and AoA combination let's assume the total lift, which is acting through the center of pressure CP which is 9m aft of the nose, is 100N.

So the balance via it's strain gauges is going to spit out 2 readings, 100N as the translational force and a pitching moment of -500Nm.

So now the JSBSim FDM has a function defined, i.e. Cm-alpha which for this combination of airspeed and AoA will return -500Nm. Plus the AERORP (often also referred to as the MRC - Moment Reference Center) has also been entered into the FDM specifying it's 7m aft of the nose in the construction frame, with in my example the origin of the construction frame being the tip of the nose.

Now the aircraft starts flying in JSBSim, and JSBSim's calculations are:

#### Case 1 - Assume the current CG is as per the drawing 7m aft of the nose

Net moment:
= -500Nm + 100N(${CG}_x - {MRC}_x$)
= -500Nm + 100N(7m - 4m)
= -500Nm + 300Nm
= **-200Nm**

#### Case 2 - Assume the current CG has now moved 1m aft, i.e. now 8m aft of the nose

Net moment:
= -500Nm + 100N(${CG}_x - {MRC}_x$)
= -500Nm + 100N(8m - 4m)
= -500Nm + 400Nm
= **-100Nm**

With the following source code line effectively doing that calculation for each timestep.

https://github.com/JSBSim-Team/jsbsim/blob/651ad3c78f68d9c0431dd5f984ff9bb8f4d24b8a/src/models/FGAerodynamics.cpp#L288

With `vMoments` being the net moment, at least for aerodynamics, and `vMomentsMRCBodyXYZ` being the sum of all the moments defined as functions returning a total moment around the `MRC\AeroRP`, often based on functions/lookup tables of wind tunnel data, and `vDXYZcg` being the difference/arm between the current cg and the AeroRP, and finally `vForces` being the sum of the forces.

**Comment by alanjt** [comment]
Sean, Agostino.

Thanks for the replies. Now that I had a quick look at the code in 
FGAerodynamics.cpp  I see that JSBsim is doing the axis transformations 
continuously.

As I was only looking at my spreadsheets  that I use to transform my 
original data to a single axis reference position I was not really 
thinking how AERORP was implemented in JSBSim.

Fortunately FGAerodynamics.cpp is easy to follow, as it has not been 
obfuscated by excessive use of OOPS. ;)

Apologies for the noise.

Whilest I have your ears do either of you know where I find a copy of 
"Nomenclature for stability coefficients, LW Bryant and SB Gates 1937,, 
R&M 1801. " as I have been unable to find it? Every report that I am 
using includes this as a reference. I do still have my old copy of 
Babister, but there may be some minor differences. I am using the this 
notation in my spreadsheets and the JSBSim .xml file, as we were still 
using it in the UK until Concorde came along, and I find it much more 
readable.

Regards

Alan

**Comment by seanmcleod70** [comment]
I don't have a handy reference to "Nomenclature for Stability Coefficients" after doing some internet searches, Gemini's AI did generate a half page summary though.

Alan, no problem in terms of your query. When I first started using JSBSim many years ago I had some queries on how the stability coefficients were defined and the relationship between AERORP and cg and how JSBSim processed it all and Dennis Linse (@djlinse) at the time provided a fairly detailed answer, which is what I used at the time to generate the diagram above.

In fact I kept his response in my notes:

> > Or have I missed something in terms of how the longitudinal moments 
> are modified and effected by the c.g. changing in JSBSim?
> 
>Yes, you've missed the underlying technique about modeling aerodynamic moments used in all flight dynamic systems (nothing to do with JSBSim).
>
>All force and moment coefficient and calculations are referenced to a single fixed point, often called the MRP (moment reference point! :-) For aircraft, this is often something like the quarter-chord point, but can be anywhere: the nose, 200 ft in front of the nose.  It doesn't matter.  Just that the reference point is known.  In fact, different coefficients can have different reference points, as long as they are fixed and known.  All forces and moments are calculated at their respective reference points and then as a final step, transferred to the current center of gravity location (as determined by the current loading, fuels, etc.) using the appropriate force and moment couples.
>
>To do it any other way would mean that each of the coefficients would have to have all of those CG corrections that you indicated.  That would be excruciating and very non-intuitive as the forces and moments aren't *caused* by the CG position.
 >
>"But...", Sean says to himself, "then how are those curves of CM changing with CG created?  You just said CM isn't a function of CG."
> 
>Think of it this way:  In a small aircraft, trimmed at a fixed angle of attack and surfaces fixed.  The aerodynamic forces and moments are all constant and don't *change* if the co-pilot stands up and walks to the back of the aircraft.  But the CG will definitely move.  The lift force that is "applied" at the 0.25 chord point now has a different moment arm with respect the new CG creating a different pitching moment with respect to the *CG*.
> 
>Also note that all figures such as this should indicate exactly *where* the CM is calculated about.  (Also note that you've included a CM-CL curve, not a CL-alpha curve)  Most don't, but they should.  For the curves you've include, the different lines are could be read as
 >
>  CM=0 about the CG at 10%MAC
>  CM=0 about the CG at 20%MAC
>  ...
> 
>Angle of attack and elevator position (among possibly other things) have changed to trim at each of these new CG locations.
> 
>OK, so that isn't as clear as a textbook, it does hit some of the high points.
> 
>Dennis
